packages/thumbnail-grid-block/src/components/Thumbnail/Thumbnail.tsx

```diff
@@ -0,0 +1,189 @@
+                                  onClick: () => setShowAltTextMenu(true),
+                                  icon: <IconSpeechBubbleQuote />,
+                              },
+                          ]
+                        : [],
+                    [
+                        {
+                            title: 'Replace with upload',
+                            icon: <IconArrowCircleUp20 />,
+                            onClick: onOpenFileDialog,
+                        },
+                        {
+                            title: 'Replace with asset',
+                            icon: <IconImageStack20 />,
+                            onClick: onAssetChooserClick,
+                        },
+                    ],
+                ].filter((item) => item.length > 0)}
+                toolbarItems={[
+                    showGrabHandle
+                        ? {
+                              onClick: () => console.log('todo: handle drag (wait for shared component)'),
+                              icon: <IconArrowMove16 />,
+                              tooltip: 'Drag to move',
+                              draggableProps,
+                          }
+                        : undefined,
+                    {
+                        icon: <IconTrashBin16 />,
+                        tooltip: 'Delete Item',
+                        onClick: () => onRemoveAsset(id, image?.id),
+                    },
+                ]}
+            >
+                <div className={thumbnailStyles.captionPositionClassNames} data-test-id="thumbnail-item">
+                    <ThumbnailImage
+                        id={id}
+                        image={image}
+                        isLoading={isLoading}
+                        isEditing={isEditing}
+                        thumbnailStyles={thumbnailStyles}
+                        defaultAltText={defaultAltText}
+                        onOpenFileDialog={onOpenFileDialog}
+                        onFilesDrop={onFilesDrop}
+                        onAssetChooserClick={onAssetChooserClick}
+                    />
+                    {(image || isEditing) && (
+                        <ThumbnailRichTextEditors
+                            id={id}
+                            title={title}
+                            isEditing={isEditing}
+                            description={description}
+                            designTokens={designTokens}
+                            updateItemWith={updateItemWith}
+                        />
+                    )}
+                </div>
+            </BlockItemWrapper>
+            <Flyout
+                trigger={<></>}
+                onOpenChange={setShowAltTextMenu}
+                fitContent
+                hug={false}
+                isOpen={showAltTextMenu}
+                placement={FlyoutPlacement.Right}
+                onConfirm={() => {
+                    updateItemWith('altText', localAltText ?? '', id);
```

Comment:
```suggestion
                    updateItemWith('altText', localAltText, id);
```

Goes with the comment before